### PR TITLE
Intermediate.ATX.2017Q1.Chris.Austin

### PR DESCRIPTION
**Pull request by oshgosh**
# Checkpoint Rubric

This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

## Checkpoint 1: Towers of Hanoi Terminal App
- [X] All tests passed: 55 points
- [X] Properly commented code: 15 points
Great job with this! (except for a couple typos ha :D)
- [X] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 15 points
7/15

Total: 92

Specific Comments:
See gtihub file comments for detailed feedback. Nice work overall, great job making things work! That's what pays the bills.

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 2: Towers of Hanoi GUI App
- [X] All tests passed: 55 points
- [X] Properly commented code: 15 points
- [X] Properly indented code: 15 points
- [X] Demonstrated effective use of jQuery: 15 points

Total: 100

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 3: Final Project Gist Blog
- [ ] Effective use of AJAX: 20 points
- [ ] Effective use of loops: 20 points
- [ ] Effective use of github API: 20 points
- [ ] Style and design: 10 points
- [ ] Properly commented code: 15 points
- [ ] Properly indented code: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

### CODE REVIEW

File: apps/01PigLatin.js [hunk 1]
```diff
@@ -9,15 +9,50 @@ var rl = readline.createInterface({
...
+  }
+  if ((word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('o');
+  }
+  if ((word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('u');
+  }
+  if ((word.indexOf('y') > -1 && word.indexOf('y') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('y');
+  }
```
Comment (zacacollier) [inline apps/01PigLatin.js:35]: Great work! try taking a look at your classmates' pull requests for some ideas on how you might write this app differently (James, Leslie and Tony had some good stuff)

File: apps/02RockPaperScissors.js [hunk 2]
```diff
@@ -9,15 +9,44 @@ var rl = readline.createInterface({
...
+
+    if (hand2 === "rock") {
+      return "Hand two wins!";
+    } else {
+      return "Hand one wins!";
+    }
+
+  } else {
+    return "Unavailable option selected."
+  }
```
Comment (zacacollier) [inline apps/02RockPaperScissors.js:43]: Nice! Input scrubbing is a great practice. You never know what kind of text your end-user might try to send through your application (e.g. malicious code)

File: apps/03TicTacToe.js [hunk 3]
```diff
@@ -24,32 +24,81 @@ function printBoard() {
...
+    || (board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn)) {
+    return true;
+  }
 }
 
 function diagonalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn)
+    || (board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn)) {
+    return true;
```
Comment (tjefferson08) [inline apps/03TicTacToe.js:45]: These win functions all look pretty good! Nice work. Did you think about returning `false` ever?

File: apps/03TicTacToe.js [hunk 4]
```diff
@@ -24,32 +24,81 @@ function printBoard() {
...
+
+function checkForTakenSquare(row, column) {
+  if (board[row][column] !== ' ') {
+    return true;
+  }
+}
+
+function checkForFullBoard() {
+  for (var i = 0; i <= 2; i++) {
+    for (var j = 0; j <= 2; j++) {
```
Comment (tjefferson08) [inline apps/03TicTacToe.js:64]: Awesome use of loops

File: apps/04TowersOfHanoi.js [hunk 5]
```diff
@@ -18,25 +18,40 @@ function printStacks() {
...
 }
-
-function isLegal() {
+// This code below checks to see if the users move was legal
+function isLegal(startStack, endStack) {
   // Your code here
+  var startBlock = stacks[startStack][stacks[startStack].length - 1];
+  var endBlock = stacks[endStack][stacks[endStack].length - 1];
+  if (startBlock < endBlock || endBlock === undefined) {
+    movePiece(startStack, endStack);
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:31]: Conceptually, this function probably shouldn't be _in charge_ of moving the piece (it's name is simply `isLegal`, so it probably shouldn't be changing the state of the board at all)

File: apps/04TowersOfHanoi.js [hunk 6]
```diff
@@ -18,25 +18,40 @@ function printStacks() {
...
+  stacks[endStack].push(stacks[startStack].pop());
 }
-
-function isLegal() {
+// This code below checks to see if the users move was legal
+function isLegal(startStack, endStack) {
   // Your code here
+  var startBlock = stacks[startStack][stacks[startStack].length - 1];
+  var endBlock = stacks[endStack][stacks[endStack].length - 1];
+  if (startBlock < endBlock || endBlock === undefined) {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:30]: Could `startBlock` ever be undefined? What would that mean as far as move legality goes?

File: apps/04TowersOfHanoi.js [hunk 7]
```diff
@@ -18,25 +18,40 @@ function printStacks() {
...
+  }
 
+}
+// This code below checks for a in - if the stack equals 4
 function checkForWin() {
   // Your code here
-
+  if (stacks.c.length === 4 || stacks.b.length === 4) {
+    return true;
+    console.log("You Won!!!");
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:45]: Can this console.log statement ever be reached? (It's right after a return statement)

File: apps/04TowersOfHanoi.js [hunk 8]
```diff
@@ -18,25 +18,40 @@ function printStacks() {
...
+function isLegal(startStack, endStack) {
   // Your code here
+  var startBlock = stacks[startStack][stacks[startStack].length - 1];
+  var endBlock = stacks[endStack][stacks[endStack].length - 1];
+  if (startBlock < endBlock || endBlock === undefined) {
+    movePiece(startStack, endStack);
+    return true;
+  } else if (startBlock > endBlock) {
+    console.log("Try Again");
+    return false;
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:35]: Notice that these return values aren't used anywhere (except directly by the tests)

File: apps/08jquery-towers-of-hanoi/script.js [hunk 9]
```diff
@@ -1,5 +1,32 @@
...
+//This below gets the document ready for jQuery
 $(document).ready(function() {
-  // Put app logic here
+  $('[data-stack]').on('click', function() {
+    //Code below initiates the game to be played with they click on it
+    if ($block === null) {
+      $block = $(this).children().last().detach();
+    } else {
+      if (
+        //this takes a block and appends it to the next
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:12]: It's probably worth mentioning the logic you put in the `if` statement here

File: apps/08jquery-towers-of-hanoi/script.js [hunk 10]
```diff
@@ -1,5 +1,32 @@
...
+  $('[data-stack]').on('click', function() {
+    //Code below initiates the game to be played with they click on it
+    if ($block === null) {
+      $block = $(this).children().last().detach();
+    } else {
+      if (
+        //this takes a block and appends it to the next 
+        $($block).data('block') < $(this).children().last().data('block') ||
+        $(this).children().length === 0) {
+        $(this).append($block);
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:15]: This is probably the stuff I'd be commenting on more so than e.g. the checkForWin comments. Think about things that are confusing for a reader (or grader 😄 )

**Pull request closed without merging**